addons/account/models/account_move.py

```diff
@@ -1422,10 +1423,11 @@ def _fetch_duplicate_supplier_reference(self, only_posted=False):
 
         move_table_and_alias = "account_move AS move"
         place_holders = {}
-        if not moves.ids:
-            # This handles the special case of a record creation in the UI which isn't searchable in the DB
+        if not moves[0].id:  # check if record is under creation/edition in UI
```

Comment:
Wasn't moves.ids checking that already ? Why didn't it work ? :o

Reply:
If the move is in edition, its `.ids` has its real id in db while its `id` is a `<NEWID>` and it has a reference to `._origin.id`.